[WIP] Tflite image segmentation example

L2zz opened the pull request:
Implement bash script for image segmentation example.
Add meson build.

Signed-off-by: Jihoon Lee <ulla4571@gmail.com>

jaeyun-jung committed:
[Android/Sample] add sample code about nnstreamer api

Add new app api-sample, this includes the sample code to use NNStreamer API.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

jaeyun-jung committed:
[Android/Build] change common tf-lite lib

1. Remove tf-lite libs in common (use android-resource repo instead)
2. Update readme to build the applications

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

myungjoo committed:
[Dist/Debian] Add explicit dependency on ninja >= 1.5

This tries to resolve:

ERROR: Could not detect Ninja v1.5 or newer

of nnstreamer-example build.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

mojunsang26 committed:
[CodeClean]unify the behavior when there is no model or label
1. set logging level to g_critical().
2. unify the error messsages.

Signed-off-by: Junsang Mo <junsang.mo@samsung.com>

mojunsang26 committed:
[CodeClean] merge get-model scripts
Delete the existing get-model scripts and merge it into model-get.sh
Identify models by arguments
Add symlink
Update meson.build script

Signed-off-by: Junsang Mo <junsang.mo@samsung.com>

Commit:
[Tizne/Native] upload sample app of nnstreamer single API

this is a simple example of nnstreamer single API

Signed-off-by: HyoungJoo Ahn <hello.ahn@samsung.com>

myungjoo committed:
Template for tensor_filter static in/out dims.

This is a template & code-gen for tensor_filter
static in/out tensor dimension subplugin.

Use deploy.sh in the given path to start
your own tensor-filter subplugin code.

To use the template:

1. Go to the template directory
```
$ cd templates/tensor_filter_subplugin
```

2. Deploy (create a new git repo for your subplugin)
```
$ deploy ${NAME} ${GIT_PATH}
$ # Then you have your template code placed at ${GIT_PATH}
$ cd ${GIT_PATH}
${GIT_PATH}$ # here, you can try gbs build.
${GIT_PATH}$ gbs build
...
${GIT_PATH}$ # You get an "empty" tensor-filter subplugin as a Tizen rpm package.
```

3. Write your subplugin with the template
```
${GIT_PATH}$ vi src/tensor_filter_subplugin.c
```

4. Now you are ready to go!

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

Commit:
[Tizne/Native] set sample application's name properly

the application's name was too general. Since it handle the pipeline API, the name should be changed.

Signed-off-by: HyoungJoo Ahn <hello.ahn@samsung.com>

myungjoo committed:
[Chmod] Add execution bit for get-model.sh

Let users execute it from the shell easily.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

mojunsang26 committed:
[CodeClean] correct the brief
remove a content about load labels in function brief.

Signed-off-by: Junsang Mo <junsang.mo@samsung.com>

L2zz commented:
I reference tensordec-boundingbox.c, it gets options like this.

https://github.com/wooksong/nnstreamer-example/blob/59bf06241eb064f17614b279865076bf40fa34a4/bash_script/example_object_detection_tensorflow_lite/gst-launch-object-detection-tflite.sh#L7

https://github.com/nnsuite/nnstreamer/blob/602296725fc770058e103dfadb494f5e8793b87d/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c#L29-L39

wooksong commented on bash_script/example_image_segmentation_tensorflow_lite/gst-launch-image-segmentation-tflite.sh:
```diff
@@ -0,0 +1,10 @@
+      t. ! tensor_converter !\
+      tensor_transform mode=arithmetic option=typecast:float32,add:0.0,div:255.0 !\
+      tensor_filter framework=tensorflow-lite model=tflite_model/deeplabv3_257_mv_gpu.tflite !\
+      tensor_decoder mode=image_segment option1=tflite !\
```
Your decoder could not handle the out of all the image segment tflite models. Need to consider it. In my opinion, you do not need option1 so far.

wooksong replied:
Or tflite/image-segment/$model_name?

I am not sure....

abcinje committed:
[Typo] Fix typo

This patch fixes typo in README.md

Signed-off-by: Injae Kang <abcinje@gmail.com>

mojunsang26 committed:
Correct typo in README.md
README.md : souce -> source
android/README.md :"a Android NDK manual" -> "an Android NDK manual"
                    "by dfault -> by default"

Signed-off-by: Junsang Mo <junsang.mo@samsung.com>

abcinje committed:
Create .gitignore to ignore build directory

I added .gitignore file to ignore build directory.

This patch resolves #74

Signed-off-by: Injae Kang <abcinje@gmail.com>

mojunsang26 committed:
update the get model guidances
change invalid get model guidances in native example sources.

Signed-off-by: Junsang Mo <junsang.mo@samsung.com>

TunaLee committed:
[Model] Add get caffe2 mobilenet_v2 script
insert contents get image-classification-caffe2 model and labels.txt.

Signed-off-by: Dongwon Lee <dongwon1103@naver.com>

L2zz committed:
[Model] Script to get tflite image segmentation model

In order to get tflite image segmentation model,
add option 'image-segmentation-tflite'.

Signed-off-by: Jihoon Lee <ulla4571@gmail.com>

L2zz committed:
[Bash/Sample] Add sample code to run tflite image segmentation

Add sample bash script which get image and paint regions of object.
Note that this example needs image file(ex. cat.png) to run.

Signed-off-by: niklasjang <niklasjang@gmail.com>
Signed-off-by: Jihoon Lee <ulla4571@gmail.com>

L2zz committed:
[Bash/Meson] Meson build for tflite image segmentation example

In order to add tflite image segmentation example to build tree,
this patch updates meson build script.

Signed-off-by: niklasjang <niklasjang@gmail.com>
Signed-off-by: Jihoon Lee <ulla4571@gmail.com>
